### DISCUSSION TITLE
Agent trying to connect from non-latest build

### DISCUSSION THREAD

**Discussion by MaksymSofer** [coder/coder#13249] [General] [answered]
Hi,
i have self-hosted coder version v2.8.3+01c6266
running in kubernetes via helm

workspaces (pods) started to throw this:
```
2024-05-12 14:25:37.043 [warn]  run exited with error ...
    error= GET http://somehost.somedomain.local/api/v2/workspaceagents/me/rpc?version=2.0: unexpected status code 403: Agent trying to connect from non-latest build.
           	Error: build is outdated
```
how do i suppose to update coder executable on pods?

**Comment by kylecarbs** [comment, marked as answer]
The agent isn't outdated here, but the token for the agent is from an old workspace build.

Try rebuilding and it should work!

**Comment by MaksymSofer** [comment]
this happen for each workspace so we need to rebuild each workspace periodically ?
Could token be updated manually somehow?

**Reply by matifali** [reply]
It's interesting that the token gets invalidated automatically. After how long does it happen?
